com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java

```diff
@@ -1425,6 +1432,89 @@ private void configurePreBuildProjectOptions() throws IOException, CompileExcept
         }
     }
 
+    private void transpileLua(IProgress monitor) throws CompileExceptionError, IOException {
+        List<ILuaTranspiler> transpilers = PluginScanner.getOrCreatePlugins("com.defold.extension.pipeline", ILuaTranspiler.class);
+        if (transpilers != null) {
+            IProgress transpilerProgress = monitor.subProgress(1);
+            transpilerProgress.beginTask("Transpiling to Lua", 1);
+            for (ILuaTranspiler transpiler : transpilers) {
+                IResource buildFileResource = getResource(transpiler.getBuildFileResourcePath());
+                if (buildFileResource.exists()) {
+                    String ext = "." + transpiler.getSourceExt();
+                    List<IResource> sources = inputs.stream()
+                            .filter(s -> s.endsWith(ext))
+                            .map(this::getResource)
+                            .collect(Collectors.toList());
+                    if (!sources.isEmpty()) {
+                        boolean useProjectDir = buildFileResource instanceof DefaultResource && sources.stream().allMatch(s -> s instanceof DefaultResource);
```

Comment:
Could you please add a comment explaining what useProjectDir is for?

Reply:
Sure!